### DISCUSSION TITLE
Question regarding shared mailboxes

### DISCUSSION THREAD

**Discussion by di6parn** [Set-OutlookSignatures/Set-OutlookSignatures#117] [Q&A] [answered]
Hello,

I know that the handling of shared mailboxes is reserved for Benefactor Circle members, but we have not yet decided if we should join, and I have a question about how it works. 

We use Office 365 and we handle our shared mailboxes in Exchange Admin Center online, where we give users their access to shared mailboxes. Nothing is done in the Outlook client for the users, we do not add the shared mailboxes as accounts. This means that today our users need to manually select the signature they need every time they send a mail from the shared mailbox. 

Will the shared mailbox feature of Set-OutlookSignatures add the desired signature automatically? 
Is this what the _SignaturesForAutomappedAndAdditionalMailboxes_ feature mentioned in the output log is for?

Cheers,
Pär

**Comment by GruberMarkus** [comment, marked as answer]
Hi Pär,

the handling of shared mailbox is not restricted to the Benefactor Circle add-on. Set-OutlookSignatures "thinks" from the client or end user perspective, there the "Outlook" in the name and not "Exchange".

The root cause of this misunderstand is that Microsoft uses the term "mailbox type" for differnt things on the server and on the client side:
- On the server side, Exchange differentiates between multiple mailbox types. The most used and well-known are user mailboxes, shared mailboxes and room/equipment mailboxes, but are many more types: Monitoring, AuditLog, Archive, PublicFolder, Arbitration, AuxAuditLog, GroupMailbox, etc.
- On the client side, Outlook also differentiates between mailbox types - but these type have nothing to do with the type of a mailbox defined on the server side. On the client side, Outlook differentiates mailboxes by the way they are added:
  - Primary mailboxes: These are mailboxes that you add with dedicated credentials. This is typically the user's own mailbox, but every other mailbox added with separate credentials is also a primary mailbox.
  - Secondary or additional mailboxes are mailboxes that are added without providing extra credentials, so they use the credential or their associated primary mailbox. These are all mailboxes that you add via "File | Account settings  | \<select mailbox\> | Change ... | More settings, Advanced | Open these additional mailboxes"
  - Automapped mailboxes are mailboxes that Outlook adds automatically because of permissions granted in Exchange in a specific way. This is the case when you grant a user (and not a group!) Read or FullAccess permission to another mailbox.  

<br>  
Now back to your question:  

The three connection types in Outlook described above all behave a bit differently in certain aspects. From a signature perspective, it is as follows:
- Primary mailboxes: Set-OutlookSignatures can detect these mailboxes, can deploy signatures for them, and can define a default signature for new emails and a default signature for replies/forwards for each of these mailboxes.
- Additional mailboxes: Set-OutlookSignatures with the Benefactor Circle add-on can detect these mailboxes and can deploy signatures for them, but it can not define a default signature. The reason for this is that Outlook simply does not support this feature. But as the signature is deployed, the user can select it manually.
- Automapped mailboxes behave just as additional mailboxes.

In the README file, this reads as follows:
- Chapter `SignaturesForAutomappedAndAdditionalMailboxes`:
  ```
  Deploy signatures for automapped mailboxes and additional mailboxes.

  Signatures can be deployed for these mailboxes, but not set as default signature due to technical restrictions in Outlook.
  
  This feature requires a Benefactor Circle license.
  ```
- Chapter `Mailboxes`
  ```
  The software only considers primary mailboxes per default, these are mailboxes added as separate accounts.

  This is the same way Outlook handles mailboxes from a signature perspective: Outlook cannot handle signatures for non-primary mailboxes (added via "Open these additional mailboxes").

  Setting the parameter SignaturesForAutomappedAndAdditionalMailboxes to true allows the software to detect automapped and additional mailboxes. Signatures can be deployed for these types of mailboxes, but they cannot be set as default signatures due to technical restrictions in Outlook.
  ```

The README chapter `"How to deploy signatures for "Send As", "Send On Behalf" etc.?` mentions additional mailboxes, too. The focus of this chapter is how to deploy signatures for mailboxes that are not visible at all in Outlook, but where a user only chooses a "from" address before sending an email.


Does this answer your question, or does it rather create new ones?

**Comment by di6parn** [comment]
Hi Markus,

Thank you for that detailed and extensive reply, I really appreciate it. And it totally answers my question. 
I realize I could have read through README better, I tried searching for "shared" but didn't find what I was looking for.

Once again, thank you, and have a nice day!

/Pär

**Reply by GruberMarkus** [reply]
My pleasure!
